mlir/lib/Dialect/Rock/Pipelines/Pipelines.cpp

```diff
@@ -221,10 +221,14 @@ void rock::buildBackendPipeline(OpPassManager &pm,
   llvmFuncPm.addPass(createCSEPass());
   llvmFuncPm.addPass(rock::createRockPrepareLLVMPass());
   if (options.compile) {
-    gpuPm.addPass(createGpuSerializeToHsacoPass(
-        options.triple, options.chip, options.features, options.optLevel,
-        options.suppressDiagnostic));
-    gpuPm.addPass(createRockCheckResidencyPass());
+    GpuROCDLAttachTargetOptions opts;
```

Comment:
Note to future us: I wonder if it might be interesting to move this target attachment early in the pipeline